cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/MultiThreadDatasetCache.java

```diff
@@ -61,7 +62,8 @@ public MultiThreadDatasetCache(final SystemDatasetInstantiator instantiator,
                                  final NamespaceId namespace,
                                  final Map<String, String> runtimeArguments,
                                  @Nullable final MetricsContext metricsContext,
-                                 @Nullable final Map<String, Map<String, String>> staticDatasets) {
+                                 @Nullable final Map<String, Map<String, String>> staticDatasets,
+                                 final MultiThreadTransactionAware<?>...multiThreadTxAwares) {
```

Comment:
added javadoc. It is possible to represent static datasets as `MultiThreadTransactionAware`s, but it requires more refactoring of the `DynamicDataset` hierarchy. We can do it in future when refactoring the class again.